mbed_flasher/flashers/FlasherMbed.py

```diff
@@ -200,14 +200,15 @@ def copy_file(self, source, destination):
 
         self.logger.debug("SHA1: %s", hashlib.sha1(aux_source).hexdigest())
 
-        if platform.system() == 'Windows':
-            self.logger.debug('copying file: "{}" to "{}"'.format(source, destination))
-            os.system("copy \"%s\" \"%s\"" % (os.path.abspath(source), destination))
-        else:
-            self.logger.debug("writing binary: %s (size=%i bytes)", destination, len(aux_source))
+        self.logger.debug("writing binary: %s (size=%i bytes)", destination, len(aux_source))
+        try:
             new_file = self.get_file(destination)
             os.write(new_file, aux_source)
+            os.fsync(new_file)
             os.close(new_file)
+        except OSError:
```

Comment:
double check that if there can be any other exceptions than OSError when above functions is called..

Reply:
Actually, in `copy_file(source, destination)` it should check the `destination` as well.
for example, in one testcase:
```
ERROR: test_copy_empty_file_linux (non_hardware.test_flash.FlashTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/mock/mock.py", line 1305, in patched
    return func(*args, **keywargs)
  File "/home/yifzuo01/workspace/mbed-flasher/test/non_hardware/test_flash.py", line 152, in test_copy_empty_file_linux
    flasher.copy_file("empty_file", "target")
  File "/home/yifzuo01/workspace/mbed-flasher/mbed_flasher/flashers/FlasherMbed.py", line 208, in copy_file
    os.fsync(new_file)
TypeError: fileno() returned a non-integer
```